coordinator/gscoordinator/launcher.py

```diff
@@ -263,18 +266,77 @@ def close_interactive_instance(self, object_id):
         )
         return process
 
+    def _launch_etcd(self):
+        self._etcd_listen_peer_port = get_free_port("localhost")
+        self._etcd_listen_client_port = get_free_port("localhost")
+
+        etcd_exec = shutil.which("etcd")
+        if not etcd_exec:
+            raise RuntimeError("etcd command not found.")
+
+        env = os.environ.copy()
+        env.update({"ETCD_MAX_TXN_OPS": "102400"})
+
+        cmd = [
+            etcd_exec,
+            "--data-dir",
+            str(self._instance_workspace),
+            "--listen-peer-urls",
+            "http://0.0.0.0:{0}".format(str(self._etcd_listen_peer_port)),
+            "--listen-client-urls",
+            "http://0.0.0.0:{0}".format(str(self._etcd_listen_client_port)),
+            "--advertise-client-urls",
+            "http://localhost:{0}".format(str(self._etcd_listen_client_port)),
+            "--initial-cluster",
+            "default=http://localhost:{0}".format(str(self._etcd_listen_peer_port)),
+            "--initial-advertise-peer-urls",
+            "http://localhost:{0}".format(str(self._etcd_listen_peer_port)),
+        ]
+        logger.info("Launch etcd with command: %s", " ".join(cmd))
+
+        process = subprocess.Popen(
+            cmd,
+            start_new_session=True,
+            cwd=os.getcwd(),
+            env=env,
+            universal_newlines=True,
+            encoding="utf-8",
+            stdin=subprocess.DEVNULL,
+            stdout=subprocess.DEVNULL,
+            stderr=subprocess.DEVNULL,
+            bufsize=1,
+        )
+
+        logger.info("Server is initializing etcd.")
+        self._etcd_process = process
+
+        start_time = time.time()
+
+        while is_free_port(self._etcd_listen_client_port):
+            time.sleep(1)
+            if (
+                self._timeout_seconds
+                and self._timeout_seconds + start_time < time.time()
+            ):
+                raise RuntimeError("Launch etcd service failed.")
```

Comment:
give detailed error message with reason. e.g., 
`Launch ectd failed due to timeout.`